### PR TITLE
sense 3494 add caching

### PR DESCRIPTION
**Pull request by philipp-leanix**
Add option to enable caching for building the docker image.

- added `enable-cache` as parameter to the action
- if caching is enabled, use buildx to cache from a registry

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -105,9 +109,16 @@ const fs = require('fs');
             const options = {stdout: (data) => core.info(data.toString()), stderror: (data) => core.error(data.toString())};
             core.info("Will now build Dockerfile at " + path + " as " + nameWithVersion);
             dockerfile_param = ((dockerfile == "")? []: ["-f", dockerfile])
-            await exec.exec('docker', ['build', '-t', nameWithVersion, ...dockerfile_param, path], options);
-            await exec.exec('docker', ['push', nameWithVersion], options);
+            if (enableCache) {
+                const nameWithCurrentVersion = name + ":" + normalisedBranch + "-" + currentVersion;
+                const cachingArgs = ["--cache-from", "type=registry,ref=" + nameWithCurrentVersion, "--cache-to", "type=inline"];
```
Comment (perbernhardt) [inline index.js:114]: Can you quickly explain how this leads to a faster build on the next run of the workspace? As far as I understood it, you need to "upload" the cache somewhere, see https://github.com/marketplace/actions/docker-layer-caching for example.

File: README.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ with:
   dockerfile: main.dockerfile  # Optional, path to a dockerfile if the filename is not "Dockerfile" (-f parameter of docker build)
   only-output-tags: true       # Optional, whether to only output the tags again and not build & push the image, by default false
   registry: acr                # Optional, registry to use, allowed values are docker-hub or acr, default is docker-hub
+  enable-cache: true           # Optional, enable caching of docker layers to registry to speed up build process, default is false
```
Comment (perbernhardt) [inline README.md:16]: If this thing works, I would prefer just enabling it by default. I cannot see why one would not want to use that? Or at least turn it around and cache by default and only disable it, e.g. to enforce package updates once a week.

Comment (philipp-leanix) [reply README.md:16]: fear of breaking existing workflows 😅

File: index.js (hunk 1 above)
Comment (perbernhardt) [reply index.js:114]: Nice, I just didn't know this feature! I like it!

File: README.md (hunk 2 above)
Comment (perbernhardt) [reply README.md:16]: Hm... So is there a risk or not?

Comment (philipp-leanix) [reply README.md:16]: There's always a risk, if somethings changed, right? But I'm fine with turning it on by default.

**Review by sebastian-hatala-leanix — APPROVED** [review]
LGTM 🥇

**Review by perbernhardt — APPROVED** [review]

**Pull request merged**